Customizing/global/plugins/Services/Cron/CronHook/UserOrguImport/classes/OrgusCleanup.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace CaT\IliasUserOrguImport;
+
+class OrgusCleanup
+{
+
+	protected $tree;
+	protected $rep_utils;
+	protected $grm;
+	protected $oc;
+
+	public function __construct($tree, $rep_utils, IliasGlobalRoleManagement $grm, Orgu\OrguConfig $oc)
+	{
+		$this->tree = $tree;
+		$this->rep_utils = $rep_utils;
+		$this->grm = $grm;
+		$this->oc = $oc;
+	}
+
+	/**
+	 * Delete all import subtree orgus, which have no user assignments and only
+	 * contain orgus without user assignments.
+	 *
+	 * @return void
+	 */
+	public function cleanupEmptyOrgus()
```

Comment:
Ich glaube der gesamte Algorithmus sieht in einer rekursiven Variante deutlich einfacher aus. Schau mal hier als Referenz:

https://github.com/conceptsandtraining/ilias-plugin-StatusMails/blob/master/classes/Orgu/ilDB.php#L30

Reply:
Ja, das ist klar, aber bei den großen Datenmengen will ich mich nicht darauf verlassen, dass wir nicht an den Stacklimit stoßen.

Reply:
Ich habe einen anderen schönen algorithmus gefunden, von dem ich glaube, dass er auch etwas fixer ist.

Reply:
Siehe das RecursiveMembersCounter -objekt. Wir werden das auch an anderer Stelle wiederverwenden könne.
Ich will nicht als renitent rüberkommen, aber ich bin kein großer fan von rekursion. Es ist immer langsamer und in meinen Augen auch schwerer zu debugen, wennauch oft etwas lesbarer. Ich versuche sie zu vermeiden, wenn es einen anderen Weg gibt, zumindest im nicht-funktionalen Kontext. Ist das OK?